chain/network/src/network_protocol/tests.rs

```diff
@@ -170,3 +176,96 @@ fn serialize_deserialize() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn serialize_deserialize_ip_addr() {
```

Comment:
This test seems redundant with the more general `serialize_deserialize_protobuf_only` above.

Reply:
I preferred adding this fine grained test. It's like saying unit tests are redundant with an end to end integration tests in place. It checks exactly the IpAddr serialize/deserialization whereas the other checks the wrapper.

Reply:
`serialize_deserialize_protobuf_only` is precisely a unit test which serializes a message, then deserializes it and checks that all fields match the original message. What kind of future mistake by a developer would this test catch which is not already covered by that test? Notice we don't have tests of this nature for all the other types of primitive fields used in network messages.

Reply:
disagree as discussed offline but this is minor so i have removed.